sdk/identity/identity/src/msal/utils.ts

```diff
@@ -222,7 +222,7 @@ export class MsalBaseUtilities {
 // transformations.ts
 
 export function publicToMsal(account: AuthenticationRecord): msalCommon.AccountInfo {
-  const [environment] = account.authority.match(/([a-z]*\.[a-z]*\.[a-z]*)/) || [];
+  const [environment] = account.authority.match(/([a-z]*\.[a-z]*\.[a-z]*)/) || [""];
```

Comment:
@KarishmaGhiya msal expect `environment` to be a `string`. would passing `""` work?

Reply:
Not sure how I missed this. Confirming this with msal team rn before we work on https://github.com/Azure/azure-sdk-for-js/issues/30684

Reply:
authority will never be null. So the "" will not work